osu.Game.Rulesets.Catch/Objects/Drawables/CaughtObjectEntry.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics.Performance;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Game.Rulesets.Catch.Objects.Drawables
+{
+    public class CaughtObjectEntry : LifetimeEntry, IHasFruitState
+    {
+        /// <summary>
+        /// Whether this caught object is stacked or dropped.
+        /// </summary>
+        public readonly CaughtObjectState State;
+
+        /// <summary>
+        /// The position of this object in relative to the catcher.
+        /// </summary>
+        public readonly Vector2 PositionInStack;
+
+        public CatchHitObject HitObject { get; }
+
+        public Bindable<Color4> AccentColour { get; } = new Bindable<Color4>();
+
+        public Bindable<bool> HyperDash { get; } = new Bindable<bool>();
+
+        public Vector2 DisplaySize { get; }
+
+        public float DisplayRotation { get; }
+
+        public Bindable<FruitVisualRepresentation> VisualRepresentation { get; } = new Bindable<FruitVisualRepresentation>();
```

Comment:
It is indeed possible to remove `VisualRepresentation` Bindable <https://github.com/ekrctb/osu/commit/e5ef1aefc0faeea03fcb4cb962042c559de4c834>.
However, it requires `IndexInBeatmap` bindable in `IHasCatchObjectState`.
Because `HitObject` referenced from `IHasObjectState` can change at anytime, a consumer cannot bind to `ObjectState.HitObject.SomeBindable` because it will become stale as soon as `ObjectState.HitObject` changes.

This is kind of bad design and I don't really like `IHasCatchObjectState` as it is very ad-hoc and there are still code duplication. Probably a better solution could be found in the future. But I don't touch it in this PR.
Also I made changes to made `CaughtObjectEntry` not implement `IHasCatchObjectState` so it is less relevant now.

Reply:
Yeah, my original comment was more of a passing remark, not something to be addressed in this PR.